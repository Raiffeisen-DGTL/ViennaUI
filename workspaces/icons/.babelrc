{
  "presets": ["@babel/env", "@babel/typescript", "@babel/react"],
  "env": {
    "esm": {
      "presets": [["@babel/env", { "modules": false }]]
    }
  }
}
