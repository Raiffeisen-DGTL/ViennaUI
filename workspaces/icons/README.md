# Библиотека иконок

###### Raiffeisenbank Design System

Коллекция иконок для использования в интерфейсах. Иконки представлены в виде React компонентов и оригинальных векторных (`*.svg`) изображений. Изображения напрямую можно использовать из пакета.
