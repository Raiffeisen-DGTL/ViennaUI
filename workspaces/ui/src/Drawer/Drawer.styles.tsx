import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const fade = getPresets('drawer.fade', {
    base: null,
    hide: null,
    show: null,
    custom: null,
});

const drawer = getPresets('drawer.window', {
    base: null,
    custom: null,
    orientation: 'orientation',
});

const content = getPresets('drawer.content', {
    base: null,
    custom: null,
});

const icon = getPresets('drawer.icon', {
    base: null,
    hover: null,
    custom: null,
});

export const Fade = styled.div<{ show?: boolean }>`
    position: fixed;
    display: flex;
    align-items: center;
    flex-direction: column;
    overflow-y: auto;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    ${fade.base}
    ${fade.custom}
    ${fade.hide}

    visibility: hidden;

    ${({ show }) =>
        show &&
        css`
            visibility: visible;
            ${fade.show}
        `}
`;

export const Box = styled.div<{
    toggle?: boolean;
    width?: string;
    height?: string;
    orientation?: 'left' | 'right' | 'top' | 'bottom';
}>`
    ${drawer.base}
    -webkit-font-smoothing: antialiased;
    overflow-y: auto;

    height: ${({ height }) => (height ? `${height}` : 'auto')};
    width: ${({ width }) => (width ? `${width}` : 'auto')};

    max-width: 100%;

    ${({ orientation }) =>
        orientation === 'left' || orientation === 'right' ? 'min-height: 100%' : 'min-width: 100%'};

    ${({ orientation }) => orientation}: -${({ orientation, width, height }) =>
    orientation === 'left' || orientation === 'right' ? (width ? `${width}` : '0') : height ? `${height}` : '0'};
    ${drawer.custom}

    ${(props) =>
        props.toggle &&
        css`
            ${drawer.orientation(props)}
        `}
`;

export const Content = styled.div`
    ${content.base}
    ${content.custom}
`;

export const CloseIcon = styled.div`
    ${icon.base}
    &:hover {
        ${icon.hover}
    }
    ${icon.custom}
`;
