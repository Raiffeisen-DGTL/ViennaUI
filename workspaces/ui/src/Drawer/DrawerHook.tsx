import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import { Drawer } from './Drawer';

let drawers: React.ReactNode[] = [];
let setDrawers: React.Dispatch<React.ReactNode[]>;
let container: HTMLDivElement;

const DrawerContainer = () => {
    const [drawers, _setDrawer] = useState<React.ReactNode[]>([]);
    setDrawers = _setDrawer;
    return <>{drawers.map((modal) => modal)}</>;
};

type DrawerHookOpen = () => void;
type DrawerHookClose = <T extends any>(data?: T) => void;

interface DrawerHook {
    open: DrawerHookOpen;
    close: DrawerHookClose;
}

export function useDrawer(): DrawerHook;
export function useDrawer(drawer: React.ReactNode, onClose: DrawerHookClose): [DrawerHookOpen, DrawerHookClose];
export function useDrawer(drawer?: any, onClose?: any): any {
    const drawerRef: any = { current: { close: null } };
    let instanse: React.ReactNode;

    const closeHandler: DrawerHookClose = (data) => {
        drawers = drawers.filter((m) => m !== instanse);
        if (typeof onClose === 'function') {
            onClose(data);
        }
    };

    const close: DrawerHookClose = (data) => {
        if (typeof drawerRef.current.close === 'function' && drawers.includes(instanse)) {
            drawerRef.current.close(data);
        }
    };

    const open: DrawerHookOpen = () => {
        if (!container) {
            container = document.createElement('div');
            document.body.appendChild(container);
            ReactDOM.render(<DrawerContainer />, container);
        }

        const state: any = { open: null };

        const handleRef = (ref) => {
            drawerRef.current = ref;
            state.open();
        };

        instanse = (
            <Drawer key={Math.random()} state={state} ref={handleRef} onClose={closeHandler}>
                {drawer}
            </Drawer>
        );

        drawers.push(instanse);
        if (setDrawers) {
            setDrawers([...drawers]);
        }
    };

    if (!drawer && !onClose) {
        return {
            set children(val) {
                drawer = val;
            },
            set onClose(val) {
                onClose = val;
            },
            set open(val) {
                this.nativeOpen = val;
            },
            get open() {
                return open;
            },
            set close(val) {
                this.nativeClose = val;
            },
            get close() {
                return close;
            },
        };
    }

    return [open, close];
}
