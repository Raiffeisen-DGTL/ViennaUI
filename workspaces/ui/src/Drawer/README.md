# Drawer

Используется для

-   Предоставления объемной информации в контексте - тексты и дополнительные настройки
-   Дополнительные действия с контентом страницы


## Импорт

```
import { Drawer, useDrawer } from 'vienna-ui';
```

## Свойства / Props

Prop | Type | Default | Description
--- | --- | --- | ---
isOpen | boolean \| undefined | false | 
closeIcon | ReactNode | <Close /> | Иконка для закрытия
noScroll | boolean \| undefined | false | Управление возможностью прокрутки всего контента или только содержимого `Drawer.Body`
orientation | "top" \| "bottom" \| "left" \| "right" \| undefined | false | Позиционирование
onClose | ((data?: any) => boolean \| void \| Promise) \| undefined | false |
onPreDispose | (() => void) \| undefined | false | Обработчик события исчезновения drawer-a
state | any | false | Управление состоянием черех хук `useDrawer`

## Использование

Компонент состоит из родительского контейнера `Drawer` и элементов контента `Drawer.Layout`, `Drawer.Head`, `Drawer.Title`, `Drawer.SubTitle`, `Drawer.Body` и `Drawer.Footer`.

```jsx
{() => {
    const [isOpen, setOpen] = React.useState(false);
    return (
        <>
            <Button onClick={() => setOpen(true)}>Show Drawer</Button>
            <Drawer isOpen={isOpen} onClose={() => setOpen(false)}>
                <Drawer.Layout>
                    <Drawer.Head>
                        <Drawer.Title>Enter SMS-code</Drawer.Title>
                        <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
                    </Drawer.Head>
                    <Drawer.Body>
                        <Demo>
                            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                            incididunt ut labore et dolore magna aliqua.
                        </Demo>
                    </Drawer.Body>
                    <Drawer.Footer>
                        <Groups>
                            <Button size='l' design='outline'>
                                Button example
                            </Button>
                            <Button size='l' design='accent' onClick={() => setOpen(false)}>
                                Close
                            </Button>
                        </Groups>
                    </Drawer.Footer>
                </Drawer.Layout>
            </Drawer>
        </>
    );
}}
```

## Признак открытого компонента
##### Свойство `isOpen`

Для того, чтобы управлять состояние открытия/закрытия Drawer-a необходимо передать соотвествующий статус `isOpen`.

 ```jsx
 <Button onClick={() => setOpen(true)}>Show Drawer</Button>
 <Drawer isOpen={isOpen} onClose={() => setOpen(false)}>
     <Drawer.Layout>
         <Drawer.Head>
             <Drawer.Title>Enter SMS-code</Drawer.Title>
             <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
         </Drawer.Head>
         <Drawer.Body>
             <Demo>
                 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                 incididunt ut labore et dolore magna aliqua.
             </Demo>
         </Drawer.Body>
         <Drawer.Footer>
             <Groups>
                 <Button size='l' design='outline'>
                     Button example
                 </Button>
                 <Button size='l' design='accent' onClick={() => setOpen(false)}>
                     Close
                 </Button>
             </Groups>
         </Drawer.Footer>
     </Drawer.Layout>
 </Drawer>
 ```

## Позиционирование
##### Свойство `orientation`

Компонент поддерживает позиционирование (свойство - `orientation`) по четырем сторонам экрана `left`, `bottom`, `right` и `top`

```jsx
<Button onClick={() => setOpen(true)}>Show Drawer</Button>
<Drawer orientation='bottom' isOpen={isOpen} onClose={() => setOpen(false)}>
    <Drawer.Layout>
        <Drawer.Head>
            <Drawer.Title>Enter SMS-code</Drawer.Title>
            <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
        </Drawer.Head>
        <Drawer.Body>
            <Demo>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                incididunt ut labore et dolore magna aliqua.
            </Demo>
        </Drawer.Body>
        <Drawer.Footer>
            <Groups>
                <Button size='l' design='outline'>
                    Button example
                </Button>
                <Button size='l' design='accent' onClick={() => setOpen(false)}>
                    Close
                </Button>
            </Groups>
        </Drawer.Footer>
    </Drawer.Layout>
</Drawer>
```


## Прокрутка длинного контента
##### Свойство `noScroll`

Возможен вариант прокрутки всего компонента или только внутреннего содержимого.
Если передается значение `true`, то для прокрутки будет доступен контент,  расположенный в `Drawer.Body`.

> Для ориентации `bottom` или `top` используйте только вариацию с прокруткой внутри компонента

```jsx
<Button onClick={() => setOpen(true)}>Show Drawer</Button>
<Drawer noScroll isOpen={isOpen} onClose={() => setOpen(false)}>
    <Drawer.Layout>
        <Drawer.Head>
            <Drawer.Title>Enter SMS-code</Drawer.Title>
            <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
        </Drawer.Head>
        <Drawer.Body>
            <Demo>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                incididunt ut labore et dolore magna aliqua.
            </Demo>
        </Drawer.Body>
        <Drawer.Footer>
            <Groups>
                <Button size='l' design='outline'>
                    Button example
                </Button>
                <Button size='l' design='accent' onClick={() => setOpen(false)}>
                    Close
                </Button>
            </Groups>
        </Drawer.Footer>
    </Drawer.Layout>
</Drawer>
```


## Иконка для закрытия
##### Свойство `closeIcon`

В качестве иконки для закрытия компонента, можно передать любой компонент.

```jsx
<Button onClick={() => setOpen(true)}>Show Drawer</Button>
<Drawer closeIcon={<>Close</>} isOpen={isOpen} onClose={() => setOpen(false)}>
    <Drawer.Layout>
        <Drawer.Head>
            <Drawer.Title>Enter SMS-code</Drawer.Title>
            <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
        </Drawer.Head>
        <Drawer.Body>
            <Demo>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                incididunt ut labore et dolore magna aliqua.
            </Demo>
        </Drawer.Body>
        <Drawer.Footer>
            <Groups>
                <Button size='l' design='outline'>
                    Button example
                </Button>
                <Button size='l' design='accent' onClick={() => setOpen(false)}>
                    Close
                </Button>
            </Groups>
        </Drawer.Footer>
    </Drawer.Layout>
</Drawer>
```

## Управление состоянием через хук
##### Свойство `state`

Управлять состоянием Drawer-a можно через хук `useDrawer` и свойство `state`, которое содержит:
```
{
  close: () => void;
  open: () => void;
}
```

```
() => {
  const state = useDrawer();
  return (
    <>
      <Button onClick={() => state.open()}>Show Drawer</Button>
      <Drawer state={state}>
        <Drawer.Layout>
          <Drawer.Head>
            <Drawer.Title>Enter SMS-code</Drawer.Title>
            <Drawer.SubTitle>Drawer SubHeader</Drawer.SubTitle>
          </Drawer.Head>
          <Drawer.Body>
            <Demo>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </Demo>
            <Select placeholder="Выберите значение">
              <Select.Option>Значение 1</Select.Option>
              <Select.Option>Значение 2</Select.Option>
              <Select.Option>Значение 3</Select.Option>
              <Select.Option>Значение 4</Select.Option>
              <Select.Option>Значение 5</Select.Option>
            </Select>
          </Drawer.Body>
          <Drawer.Footer>
            <Groups>
              <Button size="l" design="outline">
                Button example
              </Button>
              <Button size="l" design="accent" onClick={() => setOpen(false)}>
                Close
              </Button>
            </Groups>
          </Drawer.Footer>
        </Drawer.Layout>
      </Drawer>
    </>
  )
}
```
