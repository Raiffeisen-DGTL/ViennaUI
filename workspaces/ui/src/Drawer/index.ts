export * from './Drawer';
export * from './DrawerHook';
