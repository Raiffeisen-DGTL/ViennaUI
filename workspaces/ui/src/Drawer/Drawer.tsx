import React, { useCallback, useState, useEffect, ForwardRefExoticComponent, RefAttributes, useRef } from 'react';
import ReactDOM from 'react-dom';
import ReactFocusLock from 'react-focus-lock';
import { useDebounce, useWindowResize } from 'vienna.react-use';
import { Close } from 'vienna.icons';
import { Screen, ScreenSlots } from 'vienna.ui-primitives';
import { Fade, Box, CloseIcon, Content } from './Drawer.styles';
import { withTabFocusState } from '../Utils';

const ANIMATION_DURATION = 400;

export interface DrawerProps extends React.PropsWithChildren<any> {
    id?: string;
    state?: any;
    isOpen?: boolean;
    closeIcon?: React.ReactNode;
    noScroll?: boolean;
    orientation?: 'left' | 'right' | 'top' | 'bottom';
    onClose?: (data?: any) => void | boolean | Promise<boolean>;
    onPreDespose?: () => void;
    ref?: any;
}

export interface DrawerSlots extends ScreenSlots {
    Layout: typeof Screen;
}

export const Drawer: React.FC<DrawerProps> & DrawerSlots = React.forwardRef((props: DrawerProps, ref: any) => {
    const {
        id,
        children,
        state,
        onClose,
        onPreDespose,
        isOpen,
        orientation = 'right',
        closeIcon = <Close size='l' />,
    } = props;

    const [init, setInit] = useState(false);
    const [show, setShow] = useState(false);
    const [toggle, setToggle] = useState(false);

    const windowRef = useRef<any>(); // ref to DrawerBox or DrawerBox, need for getting fact width
    const [rect, setRect] = useState({ width: 'auto', height: 'auto' }); // drawer or drawer fact width, need for correct animation

    if (state && isOpen) {
        // eslint-disable-next-line no-console
        console.warn('Нельзя одновременно использовать свойства state и isOpen');
    }

    // need for styled-components accept the changes
    const initDebouncer = useDebounce(setInit, ANIMATION_DURATION);
    const toggleDebouncer = useDebounce(setToggle, 100);

    const predesposeHandler = useCallback(() => {
        setRect({ width: 'auto', height: 'auto' });
        if (typeof onPreDespose === 'function') {
            onPreDespose();
        }
    }, [onPreDespose]);

    // close animation process
    const playCloseAnimation = useCallback(() => {
        // hide drawer box
        toggleDebouncer(false, () => {
            // hide fade
            setShow(false);
        });
        // unmount
        initDebouncer(false, predesposeHandler);
    }, [toggleDebouncer, initDebouncer, predesposeHandler]);

    const close = useCallback(
        (data, force = false) => {
            if (force || typeof onClose !== 'function') {
                playCloseAnimation();
                return;
            }
            const flag = onClose(data);
            if (flag instanceof Promise) {
                flag.then(
                    (flag) => {
                        if (flag || flag === undefined) {
                            playCloseAnimation();
                        }
                    },
                    // eslint-disable-next-line no-console
                    (error) => console.error(error)
                );
                return;
            }
            if (flag === true || flag === undefined) {
                playCloseAnimation();
            }
        },
        [onClose, playCloseAnimation]
    );

    // add methods to incoming drawerHook
    if (state) {
        state.children = children;
        state.onClose = onClose;
        state.open = () => setInit(true);
        state.close = close;
    }

    const prevTarget = useRef();
    const fadeMouseDown = useCallback(
        (e) => {
            prevTarget.current = e.target;
        },
        [prevTarget]
    );

    const handleClickFade = useCallback(
        (e) => {
            // Ignore the events not coming from the "fade"
            // We don't want to close the dialog when clicking the dialog content.
            if (e.target !== e.currentTarget) {
                return;
            }

            // Make sure the event starts and ends on the same DOM element.
            if (prevTarget.current !== e.target) {
                return;
            }
            close(null);
        },
        [close]
    );

    const handleClose = useCallback(() => close(null), [close]);

    const boxClickHandler = useCallback((event) => event.stopPropagation(), []);

    const boxMouseDown = useCallback((event) => event.stopPropagation(), []);

    // ***** ANIMATION START

    // when isOpen set to true - create portal !!! DONT ADD init TO DEPS ARRAY !!!
    useEffect(() => {
        if (typeof isOpen === 'boolean') {
            if (isOpen) {
                setInit(true);
            } else if (init) {
                close(null, true);
            }
        }
    }, [isOpen, close]);

    // when portal created show fade
    const updateSize = useCallback(() => {
        if (init && windowRef.current) {
            const drawerRect = (windowRef.current as HTMLDivElement).getBoundingClientRect();
            const scrollWidth = (windowRef.current as HTMLDivElement).scrollWidth;

            if (orientation === 'left' || orientation === 'right') {
                setRect({
                    width: `${drawerRect.width}px`,
                    height: '100%',
                });
                return;
            }

            setRect({
                width: scrollWidth === window.innerWidth ? '100%' : `${scrollWidth}px`,
                height: `${drawerRect.height}px`,
            });
        }
    }, [init, orientation]);

    useWindowResize(updateSize);

    useEffect(() => {
        if (init) {
            updateSize();
            // show fade
            setShow(true);
            // show drawer - need foe correct animation
            toggleDebouncer(true);
        }
    }, [init, toggleDebouncer, updateSize]);

    // ***** ANIMATION END

    useEffect(
        () => () => {
            if (typeof ref === 'function') {
                ref({ close: (data?) => close(data), updateSize });
            } else if (ref && 'current' in ref) {
                ref.current = { close: (data?) => close(data), updateSize };
            }
        },
        [ref, close, updateSize]
    );

    // if we unmount component - despose debouncers
    useEffect(
        () => () => {
            initDebouncer.despose();
            toggleDebouncer.despose();
        },
        []
    );

    if (typeof window === 'undefined') {
        return null;
    }

    const WrappedCloseIcon = withTabFocusState((props) => (
        <CloseIcon
            tabIndex={0}
            onClick={handleClose}
            onKeyDown={(e: React.KeyboardEvent) => {
                if (e.key === 'Enter') handleClose();
            }}
            {...props}>
            {closeIcon}
        </CloseIcon>
    ));

    return (
        document &&
        ReactDOM.createPortal(
            init && (
                <ReactFocusLock returnFocus>
                    <Fade id={id} show={show} onClick={handleClickFade} onMouseDown={fadeMouseDown}>
                        <Box
                            ref={windowRef}
                            orientation={orientation}
                            toggle={toggle}
                            width={rect.width}
                            height={rect.height}
                            onClick={boxClickHandler}
                            onMouseDown={boxMouseDown}>
                            {closeIcon && <WrappedCloseIcon />}
                            <Content>{children}</Content>
                        </Box>
                    </Fade>
                </ReactFocusLock>
            ),
            document.body
        )
    );
}) as ForwardRefExoticComponent<DrawerProps & RefAttributes<any>> & DrawerSlots;

Drawer.displayName = 'Drawer';

Drawer.Layout = Screen;
Drawer.Head = Screen.Head;
Drawer.Title = Screen.Title;
Drawer.SubTitle = Screen.SubTitle;
Drawer.Body = Screen.Body;
Drawer.Footer = Screen.Footer;

export default Drawer;
