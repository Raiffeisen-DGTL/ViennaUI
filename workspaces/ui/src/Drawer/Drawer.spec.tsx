import React from 'react';
import { Close } from 'vienna.icons';
import { Fade, Box, Content, CloseIcon } from './Drawer.styles';

test('Drawer /w as drawer left', () => {
    const snap = snapshot.render(
        <Fade>
            <Box orientation='left' width='0px' height='0px'>
                <CloseIcon>
                    <Close />
                </CloseIcon>
                <Content>content</Content>
            </Box>
        </Fade>
    );
    expect(snap).toMatchSnapshot();
});

test('Drawer /w as drawer left show', () => {
    const snap = snapshot.render(
        <Fade show>
            <Box toggle orientation='left' width='0px' height='0px'>
                <CloseIcon>
                    <Close />
                </CloseIcon>
                <Content>content</Content>
            </Box>
        </Fade>
    );
    expect(snap).toMatchSnapshot();
});

test('Drawer /w as drawer top', () => {
    const snap = snapshot.render(
        <Fade>
            <Box orientation='top' width='0px' height='0px'>
                <CloseIcon>
                    <Close />
                </CloseIcon>
                <Content>content</Content>
            </Box>
        </Fade>
    );
    expect(snap).toMatchSnapshot();
});

test('Drawer /w as drawer right', () => {
    const snap = snapshot.render(
        <Fade>
            <Box orientation='right' width='0px' height='0px'>
                <CloseIcon>
                    <Close />
                </CloseIcon>
                <Content>content</Content>
            </Box>
        </Fade>
    );
    expect(snap).toMatchSnapshot();
});

test('Drawer /w as drawer bottom', () => {
    const snap = snapshot.render(
        <Fade>
            <Box orientation='bottom' width='0px' height='0px'>
                <CloseIcon>
                    <Close />
                </CloseIcon>
                <Content>content</Content>
            </Box>
        </Fade>
    );
    expect(snap).toMatchSnapshot();
});
