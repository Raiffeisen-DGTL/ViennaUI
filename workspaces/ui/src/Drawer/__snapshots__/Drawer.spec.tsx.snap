// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer /w as drawer bottom 1`] = `
<div
  class="Drawerstyles__Fade-sc-34en6x-0 cIwlCK"
>
  <div
    class="Drawerstyles__Box-sc-34en6x-1 oNlZK"
    height="0px"
    orientation="bottom"
    width="0px"
  >
    <div
      class="Drawerstyles__CloseIcon-sc-34en6x-3 eSIqqc"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="Drawerstyles__Content-sc-34en6x-2 hyZwjZ"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Drawer /w as drawer left 1`] = `
<div
  class="Drawerstyles__Fade-sc-34en6x-0 cIwlCK"
>
  <div
    class="Drawerstyles__Box-sc-34en6x-1 bskFvj"
    height="0px"
    orientation="left"
    width="0px"
  >
    <div
      class="Drawerstyles__CloseIcon-sc-34en6x-3 eSIqqc"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="Drawerstyles__Content-sc-34en6x-2 hyZwjZ"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Drawer /w as drawer left show 1`] = `
<div
  class="Drawerstyles__Fade-sc-34en6x-0 fUpiCd"
>
  <div
    class="Drawerstyles__Box-sc-34en6x-1 dKuBiN"
    height="0px"
    orientation="left"
    width="0px"
  >
    <div
      class="Drawerstyles__CloseIcon-sc-34en6x-3 eSIqqc"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="Drawerstyles__Content-sc-34en6x-2 hyZwjZ"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Drawer /w as drawer right 1`] = `
<div
  class="Drawerstyles__Fade-sc-34en6x-0 cIwlCK"
>
  <div
    class="Drawerstyles__Box-sc-34en6x-1 iRIEdk"
    height="0px"
    orientation="right"
    width="0px"
  >
    <div
      class="Drawerstyles__CloseIcon-sc-34en6x-3 eSIqqc"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="Drawerstyles__Content-sc-34en6x-2 hyZwjZ"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Drawer /w as drawer top 1`] = `
<div
  class="Drawerstyles__Fade-sc-34en6x-0 cIwlCK"
>
  <div
    class="Drawerstyles__Box-sc-34en6x-1 dAAfTG"
    height="0px"
    orientation="top"
    width="0px"
  >
    <div
      class="Drawerstyles__CloseIcon-sc-34en6x-3 eSIqqc"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="Drawerstyles__Content-sc-34en6x-2 hyZwjZ"
    >
      content
    </div>
  </div>
</div>
`;
